core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -573,8 +610,43 @@ public TableMetadata replaceSortOrder(SortOrder newOrder) {
 
     return new TableMetadata(null, formatVersion, uuid, location,
         lastSequenceNumber, System.currentTimeMillis(), lastColumnId, currentSchemaId, schemas, defaultSpecId, specs,
-        lastAssignedPartitionId, newOrderId, builder.build(), properties, currentSnapshotId, snapshots, snapshotLog,
-        addPreviousFile(file, lastUpdatedMillis));
+        lastAssignedPartitionId, newOrderId, builder.build(), defaultRowKeyId, rowKeys,
+        properties, currentSnapshotId, snapshots, snapshotLog, addPreviousFile(file, lastUpdatedMillis));
+  }
+
+  public TableMetadata updateRowKey(RowKey newKey) {
```

Comment:
We will need to introduce a public iceberg table API to update the RowKey specification, right ?  Maybe we could file a separate issue to address this, we can publish the interface in that issue.

Reply:
Yes, I have a separate PR for that after this is out.